Detections/SecurityEvent/UserPrincipalNameAssignedToUserAccount.yaml

```diff
@@ -26,16 +26,23 @@ query: |
     | parse EventData with * 'ObjectDN">' ObjectDN "<" *
     | parse EventData with * 'AttributeValue">' AttributeValue "<" *
     | summarize StartTime = min(TimeGenerated), EndTime = max(TimeGenerated) by Computer, SubjectAccount, ObjectDN, AttributeValue
+    | extend HostName = tostring(split(Computer, ".")[0]), DomainIndex = toint(indexof(Computer, '.'))
+    | extend HostNameDomain = iff(DomainIndex != -1, substring(Computer, DomainIndex + 1), Computer)
+    | extend AccountName = tostring(split(SubjectAccount, "@")[0]), AccountUPNSuffix = tostring(split(SubjectAccount, "@")[1])
 entityMappings:
   - entityType: Account
     fieldMappings:
-      - identifier: FullName
-        columnName: SubjectAccount
+      - identifier: Name
```

Comment:
SubjectAccount is not a UPN in this case. It is a Name + NTDomain.  There should already be fields available for SubjectUserName and SubjectDomainName, you can include them in the Summarize so you can then map them properly.